### PR TITLE
EVG-16641: fix number input and schema typing

### PR DESCRIPTION
**Pull request by sophstad**
EVG-16641

### Description 
- Fix bug where number inputs would require an extra backspace to clear entirely
- Lots of other small fixes related to typing — this
- Add missing `"ui:orderable": false` to arrays

### Testing 
- Update unit and integration tests

### CODE REVIEW

File: src/pages/projectSettings/tabs/GeneralTab/getFormSchema.tsx [hunk 1]
```diff
@@ -147,20 +148,18 @@ export const getFormSchema = (
...
-                  ),
-                ],
-                enum: [
-                  ...validDefaultLoggers,
-                  ...insertIf(
-                    projectType === ProjectType.AttachedProject,
-                    null
-                  ),
+                type: "string" as "string",
+                oneOf: [
```
Comment (sophstad) [inline src/pages/projectSettings/tabs/GeneralTab/getFormSchema.tsx:152]: The `oneOf` and `enum` patterns are different ways to do the same thing; use `oneOf` to align more with implementation elsewhere

File: src/pages/projectSettings/tabs/GithubCommitQueueTab/transformers.ts [hunk 2]
```diff
@@ -58,41 +68,41 @@ export const gqlToForm: GqlToFormFunction<FormState> = (
...
-      gitTagVersionsEnabled: projectRef.gitTagVersionsEnabled,
+      gitTagVersionsEnabled,
       users: {
-        gitTagAuthorizedUsersOverride: override(
-          projectRef.gitTagAuthorizedUsers
-        ),
-        gitTagAuthorizedUsers: projectRef.gitTagAuthorizedUsers,
+        gitTagAuthorizedUsersOverride:
+          projectType !== ProjectType.AttachedProject ||
+          !!gitTagAuthorizedUsers,
```
Comment (sophstad) [inline src/pages/projectSettings/tabs/GithubCommitQueueTab/transformers.ts:87]: Update this logic to correctly allow users to override repo authorized users.

File: src/pages/projectSettings/tabs/PluginsTab/types.ts [hunk 3]
```diff
@@ -9,19 +9,17 @@ export interface FormState {
...
-    }> | null;
+    ticketSearchProjects: Array<{ searchProject: string }>;
     ticketCreateProject: {
-      createProject: string | null;
+      createProject: string;
     };
     fileTicketWebhook: {
-      endpoint: string | null;
-      secret: string | null;
-    } | null;
```
Comment (sophstad) [inline src/pages/projectSettings/tabs/PluginsTab/types.ts:24]: Same as other type changes; due to the way `PluginTab/transformers.ts` is implemented, these fields are never actually `null`.

File: src/pages/projectSettings/tabs/GithubCommitQueueTab/transformers.ts (hunk 2 above)
Comment (malikchaya2) [reply src/pages/projectSettings/tabs/GithubCommitQueueTab/transformers.ts:87]: Can you please explain this change a bit more?

Comment (malikchaya2) [reply src/pages/projectSettings/tabs/GithubCommitQueueTab/transformers.ts:87]: ah nvm, I think I understand

**Review by malikchaya2 — APPROVED** [review]
Nice work! And thank you for adding the PR comments, they were very helpful.

File: cypress/integration/projectSettings/project_settings.ts [hunk 4]
```diff
@@ -457,45 +461,57 @@ describe("Project Settings when defaulting to repo", () => {
...
+    it("Does not default to repo value for display name", () => {
+      cy.dataCy("display-name-input").should("not.have.attr", "placeholder");
+    });
 
-  it("Shows a navigation warning modal when clicking on a header link", () => {
-    cy.get("a[href='/user/patches']").click();
-    cy.dataCy("navigation-warning-modal").should("be.visible");
-    cy.get("body").type("{esc}");
-  });
+    it("Shows a navigation warning modal when clicking on a header link", () => {
```
Comment (khelif96) [inline cypress/integration/projectSettings/project_settings.ts:495]: nit is this behavior exclusive to clicking a header link or navigating away in general?
if the latter maybe just giving it a more generic test name.

File: cypress/integration/projectSettings/project_settings.ts [hunk 5]
```diff
@@ -457,45 +461,57 @@ describe("Project Settings when defaulting to repo", () => {
...
+      cy.get("a[href='/user/patches']").click();
```
Comment (khelif96) [inline cypress/integration/projectSettings/project_settings.ts:496]: nit. `cy.contains("My Patches")`

**Review by khelif96 — APPROVED** [review]
LGTM just 2 nits

**Pull request merged**